.github/workflows/pr.yaml

```diff
@@ -0,0 +1,50 @@
+name: CI for PRs
+
+on:
+  pull_request:
+    paths-ignore:
+      - "**.md"
+    branches:
+      - main
+
+jobs:
+  ci_checks:
+    name: CI Checks
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        package:
+          - packages/yx_scope
+          - packages/yx_scope_flutter
+          - packages/yx_scope_linter
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: '3.27.0'
```

Comment:
the most recent is `3.27.1`, should we move it to the matrix to test against 2+ recent major releases?